cpp/ql/src/experimental/Security/CWE/CWE-552/DangerousUseRename.ql

```diff
@@ -0,0 +1,66 @@
+/**
+ * @name Dangerous use rename.
+ * @description If the rename function did not work correctly, simply overwriting the destination file can lead to errors.
+ * @kind problem
+ * @id cpp/dangerous-use-rename
+ * @problem.severity warning
+ * @precision medium
+ * @tags correctness
+ *       security
+ *       external/cwe/cwe-552
+ */
+
+import cpp
+import semmle.code.cpp.valuenumbering.GlobalValueNumbering
+
+/** Holds if the function argument is used in opening the file for reading. */
+predicate findFileForRead(FunctionCall fc, FunctionCall fc1) {
+  fc1.getTarget().hasGlobalOrStdOrBslName(["open", "fopen"]) and
+  globalValueNumber(fc1.getArgument(0)) = globalValueNumber(fc.getArgument(0)) and
+  fc1.getArgument(1).getValue() = "r" and
+  fc.getASuccessor+() = fc1
+}
+
+/** Holds if the function argument is used in opening the file for writing. */
+predicate findFileForWrite(FunctionCall fc, FunctionCall fc1) {
+  fc1.getTarget().hasGlobalOrStdOrBslName(["open", "fopen"]) and
+  globalValueNumber(fc1.getArgument(0)) = globalValueNumber(fc.getArgument(1)) and
+  fc1.getNumberOfArguments() = 2 and
+  fc1.getArgument(1).getValue() = "w" and
```

Comment:
What about cases where the `mode` argument consists of multiple characters?

Reply:
It is important for me to find a job with one flag.
I want to avoid false positives when the files is then manipulated.